discourse-doctor

```diff
@@ -0,0 +1,403 @@
+#!/usr/bin/env bash
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $DIR
+
+LOG_FILE="$DIR/discourse-debug.txt"
+rm -f $LOG_FILE
+echo DISCOURSE DOCTOR `date` | tee -a $LOG_FILE
+
+##
+## Make sure only root can run our script
+##
+check_root() {
+  if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root. Please sudo or log in as root first." 1>&2
+    exit 1
+  fi
+}
+
+##
+## Check whether a connection to HOSTNAME ($1) on PORT ($2) is possible
+##
+connect_to_port () {
+  HOST="$1"
+  PORT="$2"
+  VERIFY=`date +%s | sha256sum | base64 | head -c 20`
+  echo -e "HTTP/1.1 200 OK\n\n $VERIFY" | nc -w 4 -l -p $PORT >/dev/null 2>&1 &
+  if curl --proto =http -s $HOST:$PORT --connect-timeout 3 | grep $VERIFY >/dev/null 2>&1
+  then
+      return 0
+  else
+    curl --proto =http -s localhost:$PORT >/dev/null 2>&1
+    return 1
+  fi
+}
+
+check_IP_match () {
+  HOST="$1"
+  echo
+  echo Checking your domain name . . .
+  if connect_to_port $HOST 443
+  then
+      echo
+      echo "Connection to $HOST succeeded."
+  else
+    echo WARNING:: This server does not appear to be accessible at $HOST:443.
+    echo
+    if connect_to_port $HOST 80
+    then
+	echo A connection to port 80 succeeds, however.
+	echo This suggests that your DNS settings are correct,
+	echo but something is keeping traffic to port 443 from getting to your server.
+	echo Check your networking configuration to see that connections to port 443 are allowed.
+    else
+      echo "A connection to http://$HOST (port 80) also fails."
+      echo
+      echo This suggests that $HOST resolves to the wrong IP address
+      echo or that traffic is not being routed to your server.
+    fi
+    echo
+    echo Google: \"open ports YOUR CLOUD SERVICE\" for information for resolving this problem.
+    echo
+    echo You should probably answer \"n\" at the next prompt and disable Let\'s Encrypt.
+    echo
+    echo This test might not work for all situations,
+    echo so if you can access Discourse at http://$HOST, you might try anyway.
+    sleep 3
+  fi
+}
+
+##
+## Do we have docker?
+##
+check_docker_version () {
+  echo  -e "\n\n==================== DOCKER INFO ====================" | tee -a $LOG_FILE
+  docker_path=`which docker.io || which docker`
+  if [ -z $docker_path ]; then
+    echo "Docker is not installed. Have you installed Discourse at all?"
+    echo "Perhaps you're looking for ./discourse-setup ."
+    echo "There is no point in continuing."
+    exit
+  else
+    echo -n "DOCKER VERSION: " | tee -a $LOG_FILE
+    docker --version | tee -a  $LOG_FILE
+    echo -e "\nDOCKER PROCESSES" | tee -a $LOG_FILE
+    sudo docker ps -a | tee -a $LOG_FILE
+  fi
+}
+
+##
+## What are we running on
+##
+check_OS() {
+  echo "----------------------------------------"
+  echo -n "OS:"  |tee -a $LOG_FILE
+  echo `uname -s` |tee -a  $LOG_FILE
+}
+
+##
+## OS X available memory
+##
+check_osx_memory() {
```

Comment:
AFAIK discourse-doctor is for supported docker installs and discourse officially supports only 64 bit Linux compatible with Docker . SO I'm unsure if we should bother with macOS stuff here.

Reply:
I totally agree, and I don't know how that OSX code snuck in to `discourse-setup`, but tons of people (for reasons that I don't understand) try to run Discourse on a Mac, and regularly seek help, so we may as well leave it in.